front_end/audits2/Audits2Panel.js

```diff
@@ -1,43 +1,242 @@
+  }
+
+  /**
+   * @param {string} bloburl
+   * @return {!Element}
+   */
+  _createIframe(bloburl) {
+    var iframeContainer = createElementWithClass('div', 'iframe-container');
+    var iframe = iframeContainer.createChild('iframe', 'fill');
+    iframe.setAttribute('sandbox', 'allow-scripts allow-popups-to-escape-sandbox allow-popups');
+    iframe.src = bloburl;
+    return iframeContainer;
+  }
+
+  /**
+   * @param {string} url
+   * @param {string} timestamp
+   * @return {!Element}
+   */
+  _createResultsBar(url, timestamp) {
+    var elem = createElementWithClass('div', 'results-bar hbox');
+    elem.createChild('div', 'logo-small');
+
+    var summaryElem = elem.createChild('div', 'summary');
+    var reportFor = summaryElem.createChild('span');
+    reportFor.innerHTML = `Report for <b>${url}</b>`;
+    var timeElem = summaryElem.createChild('span');
+    timeElem.textContent =
+        `Generated at ${new Date(timestamp).toLocaleDateString()} ${new Date(timestamp).toLocaleTimeString()}`;
+
+    var newAuditButton = createTextButton(Common.UIString('New Audit'), this._reset.bind(this), 'new-audit audit-btn');
+    elem.appendChild(newAuditButton);
+    return elem;
+  }
+};
+
+Audits2.ProtocolService = class extends Common.Object {
+  constructor() {
+    super();
+    this._rawConnection = undefined;
+    this._backend = undefined;
+    this._backendPromise = undefined;
+    this._status = undefined;
+  }
+
+  /**
+   * @return {!Promise<undefined>}
+   */
+  attach() {
+    return SDK.targetManager.interceptMainConnection(this._dispatchProtocolMessage.bind(this))
+        .then(rawConnection => {
+          this._rawConnection = rawConnection;
+        });
+  }
+
+  /**
+   * @param {string} inspectedURL
+   * @return {!Promise<!Object|undefined>}
+   */
+  startLighthouse(inspectedURL) {
+      return this._send('start', {url: inspectedURL});
+  }
+
+  /**
+   * @return {!Promise<!Object|undefined>}
+   */
+  detach() {
+    return new Promise((resolve, reject) =>
+       this._send('stop').then(() => {
+        this._backend.dispose();
+        delete this._backend;
+        delete this._backendPromise;
+        return this._rawConnection.disconnect().then(resolve);
+      })
+    );
+  }
+
+  /**
+   *  @param {function (string): undefined} callback
+   */
+  registerStatusCallback(callback) {
+    this._status = callback;
+  }
+
+  /**
    * @param {string} message
    */
   _dispatchProtocolMessage(message) {
     this._send('dispatchProtocolMessage', {message: message});
   }
 
-  _stop() {
-    this._send('stop').then(() => {
-      this._rawConnection.disconnect();
-      this._backend.dispose();
-      delete this._backend;
-      delete this._backendPromise;
+  _initWorker() {
+    this._backendPromise = Services.serviceManager.createAppService('audits2_worker', 'Audits2Service', false);
+    this._backendPromise.then(backend => {
+      /** @type {!ServiceManager.Service} */
```

Comment:
> `WARNING - Bad type annotation. Unknown type ServiceManager.Service`

usually this means closure can't see the declaration of `ServiceManager.Service`, wherever that is. Is the audits 2 panel being compiled separately from whatever file that's in?